sisl/io/siesta/out.py

```diff
@@ -878,5 +879,416 @@ def MDstep_dataframe(scf):
+        Pass `Opt.ANY` as arguments to `iscf` for cases *A* and *D*.
+
+        Parameters
+        ----------
+        name: {"voronoi", "hirshfeld"}
+            the name of the charges that you want to read
+        iscf: int or Opt, optional
+            index (0-based) of the scf iteration you want the charges for.
+            If the enum specifier `Opt.ANY` or `Opt.ALL` are used, then
+            the returned quantities depend on what is present.
+        imd: int or Opt, optional
+            index (0-based) of the md step you want the charges for.
+            If the enum specifier `Opt.ANY` or `Opt.ALL` are used, then
+            the returned quantities depend on what is present.
+        key_scf : str, optional
+            the key lookup for the scf iterations (a ":" will automatically be appended)
+        as_dataframe: boolean, optional
+            whether charges should be returned as a pandas dataframe.
+
+        Returns
+        -------
+        numpy.ndarray
+            if a specific MD+SCF index is requested (or special cases where output is
+            not complete)
+        list of numpy.ndarray
+            if one of `iscf` or `imd` is None and the output contains the other.
+        pandas.DataFrame
+            if `as_dataframe` is requested. Here the returned data frames may be
+            empty; see Notes above. The dataframe will have multi-indices if multiple
+            SCF or MD steps are requested.
+        """
+        if not hasattr(self, 'fh'):
+            with self:
+                return read_charge(self, name, iscf, imd, key_scf, as_dataframe)
+        namel = name.lower()
+        if as_dataframe:
+            import pandas as pd
+            def _empty_charge():
+                # build a fake dataframe with no indices
+                return pd.DataFrame(index=pd.Index([], name="atom", dtype=np.int32),
+                                    dtype=np.float32)
+        else:
+            pd = None
+            def _empty_charge():
+                # return for single value with nan values
+                return _a.arrayf([[None]])
+
+        # define helper function for reading voronoi+hirshfeld charges
+        def _voronoi_hirshfeld_charges():
+            """ Read output from Voronoi/Hirshfeld charges """
+            nonlocal pd
+
+            # Expecting something like this:
+            # Voronoi Atomic Populations:
+            # Atom #     dQatom  Atom pop         S        Sx        Sy        Sz  Species
+            #      1   -0.02936   4.02936   0.00000  -0.00000   0.00000   0.00000  C
+
+            # Define the function that parses the charges
+            def _parse_charge(line):
+                atom_idx, *vals, symbol = line.split()
+                return list(map(float, vals))
+
+            # first line is the header
+            header = (self.readline()
+                      .replace("Qatom", "q") # Qatom in 4.1, dQatom in master
```

Comment:
Does this result in "q" for 4.1 and "dq" for master? I thought it was better to unify them with the same names. Otherwise sisl code needs to be different depending on the version of SIESTA that generated the log :(

Reply:
True, I missed this, will fix this.

The point was that that column refers to `dq`, so I want the column to be named `dq`, and the population to be named `e`.